neuraxle/distributed/streaming.py

```diff
@@ -0,0 +1,511 @@
+    def _transform_data_container(self, data_container: DataContainer, context: ExecutionContext) -> DataContainer:
+        """
+        Transform data container
+
+        :param data_container: data container to transform.
+        :type data_container: DataContainer
+        :param context: execution context
+        :type context: ExecutionContext
+        :return: data container
+        """
+        data_container_batches = data_container.convolved_1d(stride=self.batch_size, kernel_size=self.batch_size)
+        n_batches = self.get_n_batches(data_container)
+
+        queue_joiner = QueueJoiner(n_batches=n_batches)
+        self.connect_queue_joiner(queue_joiner)
+
+        for i, (name, step) in enumerate(self):
+            step.start(context)
+
+        for data_container_batch in data_container_batches:
+            self.notify_new_batch_to_process(data_container=data_container_batch, queue_joiner=queue_joiner)
+
+        return queue_joiner.join(original_data_container=data_container)
+
+    def _did_transform(self, data_container: DataContainer, context: ExecutionContext) -> DataContainer:
+        """
+        Stop all of the workers after transform.
+
+        :param data_container: data container
+        :type data_container: DataContainer
+        :param context: execution context
+        :type context: ExecutionContext
+        :return: data container
+        :rtype: DataContainer
+        """
+        for name, step in self:
+            step.stop()
+
+        return self.data_joiner.handle_transform(data_container, context)
+
+    @abstractmethod
+    def get_n_batches(self, data_container):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def connect_queue_workers(self):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def connect_queue_joiner(self, queue_joiner):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def notify_new_batch_to_process(self, data_container, queue_joiner):
+        raise NotImplementedError()
+
+
+class SequentialQueuedPipeline(BaseQueuedPipeline):
+    """
+    Using :class:`QueueWorker`, run all steps sequentially even if they are in separate processes or threads.
+
+    .. seealso::
+        :class:`QueueWorker`,
+        :class:`BaseQueuedPipeline`,
+        :class:`ParallelQueuedPipeline`,
+        :class:`QueueJoiner`,
+        :class:`Observer`,
+        :class:`Observable`
+    """
+
+    def get_n_batches(self, data_container):
+        return data_container.get_n_batches(self.batch_size)
+
+    def connect_queue_workers(self):
+        for i, (name, step) in enumerate(self):
+            if i != 0:
+                self[i - 1].subscribe(step)
+
+    def connect_queue_joiner(self, queue_joiner: 'QueueJoiner'):
+        self[-1].subscribe(queue_joiner)
+
+    def notify_new_batch_to_process(self, data_container: DataContainer, queue_joiner: 'QueueJoiner'):
+        data_container = data_container.set_summary_id(data_container.hash_summary())
```

Comment:
in docstring of pipeline, tell that workers might disorder batches
make it clear in docstring that queue joiner reorders the processed batches at the end....